File: Monitoring_Web/Monitoring_Web/ClientApp/src/app/_model/sys-integration-api.ts
export interface IntegrationAPI {
    nodeID: number | null;
    nodeName: string | null;
}export interface CreateIntegrationAPIRequest {
    environmentID: number;
    machineName: | null;
    healthMeasurementKey: string | null;
    appid: string | null;
    domain_SystemHealth: string | null;
    nodeType: number;
    isActive: boolean | null;
    serviceList: string | null;
    isDefaultNode: boolean | null;
}export interface UpdateIntegrationAPIRequest {
    id: number;
    environmentId: number;
    environmentName: string| null;
    machineName: string;
    healthMeasurementKey: string | null;
    appid: string | null;
    domain_SystemHealth: string | null;
    isActive: boolean | null;
    createDate: string | null;
    isCheck:boolean | null;
}export interface IntegrationAPIRequest {
    machineName: string;
    isActive: string;
    PageIndex: number;
    PageSize: number;
}

File: Monitoring_Web/Monitoring.Model/Entity/Sys_Integration_API.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Monitoring.Model.Entity
{
    [Table("Sys_Integration_API")]
    public partial class Sys_Integration_API
    {
        [Key]
        public int ID { get; set; }
        public int? EnvironmentID { get; set; }
        [StringLength(500)]
        public string? MachineName { get; set; }
        public string? HealthMeasurementKey { get; set; }
        public string? Appid { get; set; }
        [StringLength(500)]
        [Unicode(false)]
        public string? domain_SystemHealth { get; set; }
        public Int16? NodeType { get; set; }
        public bool? IsActive { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? CreateDate { get; set; }
        [StringLength(4000)]
        public string? ServiceList { get; set; }
        public bool? IsDefaultNode { get; set; }
    }
}
